File: Easy/Logic Gates/My Published Solutions/Solution FP.java
import java.util.*;

class Solution {

    public static void main(String args[]) {

        Scanner in = new Scanner(System.in);
        int n = in.nextInt(); // Number of inputs
        int m = in.nextInt(); // Number of outputs
        Map <String, String> inputs = new HashMap<String, String>(); // Hashmap for the input (name, signal)

        for (int i = 0; i < n; i++) { // Put the values of the input in the Hashmap
            String inName = in.next();
            String inSignal = in.next();
            inputs.put(inName, inSignal);
        }

        for (int i = 0; i < m; i++) { // Create a different output per loop iteration
            String outName = in.next();
            String type = in.next();
            String inName1 = in.next();
            String inName2 = in.next();
            String outSignal = "";

            if (type.equals("AND")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        if (inputs.get(inName1).charAt(j) == '-') {
                            outSignal += "-";
                        } else {
                            outSignal += "_";
                        }
                    } else {
                        outSignal += "_";
                    }
                }
            }

            if (type.equals("OR")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        if (inputs.get(inName1).charAt(j) == '-') {
                            outSignal += "-";
                        } else {
                            outSignal += "_";
                        }
                    } else {
                        outSignal += "-";
                    }
                }
            }

            if (type.equals("XOR")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        outSignal += "_";
                    } else {
                        outSignal += "-";
                    }
                }
            }

            if (type.equals("NAND")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        if (inputs.get(inName1).charAt(j) == '-') {
                            outSignal += "_";
                        } else {
                            outSignal += "-";
                        }
                    } else {
                        outSignal += "-";
                    }
                }
            }

            if (type.equals("NOR")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        if (inputs.get(inName1).charAt(j) == '-') {
                            outSignal += "_";
                        } else {
                            outSignal += "-";
                        }
                    } else {
                        outSignal += "_";
                    }
                }
            }

            if (type.equals("NXOR")) {
                for (int j = 0; j < inputs.get(inName1).length(); j++) {
                    if (inputs.get(inName1).charAt(j) == inputs.get(inName2).charAt(j)) {
                        outSignal += "-";
                    } else {
                        outSignal += "_";
                    }
                }
            }

            System.out.println(outName + " " + outSignal);
        }
    }
}

File: Easy/Logic Gates/My Published Solutions/Solution FP.js
const n = parseInt(readline());
const m = parseInt(readline());
var inputs = new Map()

for (let i = 0; i < n; i++) {
    var inp = readline().split(' ');
    const in_name = inp[0];
    const in_signal = inp[1];
    inputs.set(in_name, in_signal)
}

for (let i = 0; i < m; i++) {
    var out = readline().split(' ');
    const out_name = out[0];
    const type = out[1];
    const in_name1 = out[2];
    const in_name2 = out[3];
    var out_signal = "";

    if(type == "AND"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                if(inputs.get(in_name1).charAt(j) == '-') {
                    out_signal += '-';
                } else {
                    out_signal += '_';
                }
            } else {
                out_signal += '_';
            }
        }
    }

    if(type == "OR"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                if(inputs.get(in_name1).charAt(j) == '-') {
                    out_signal += '-';
                } else {
                    out_signal += '_';
                }
            } else {
                out_signal += '-';
            }
        }
    }

    if(type == "XOR"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                out_signal += '_'
            } else {
                out_signal += '-';
            }
        }
    }

    if(type == "NAND"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                if(inputs.get(in_name1).charAt(j) == '-') {
                    out_signal += '_';
                } else {
                    out_signal += '-';
                }
            } else {
                out_signal += '-';
            }
        }
    }

    if(type == "NOR"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                if(inputs.get(in_name1).charAt(j) == '-') {
                    out_signal += '_';
                } else {
                    out_signal += '-';
                }
            } else {
                out_signal += '_';
            }
        }
    }

    if(type == "NXOR"){
        for(let j = 0; j < inputs.get(in_name1).length; j++) {
            if(inputs.get(in_name1).charAt(j) == inputs.get(in_name2).charAt(j)){
                out_signal += '-';
            } else {
                out_signal += '_';
            }
        }
    }

    console.log(out_name + " " + out_signal);
}
